FunctionBar.c

```diff
@@ -92,6 +92,28 @@ int FunctionBar_draw(const FunctionBar* this) {
    return FunctionBar_drawExtra(this, NULL, -1, false);
 }
 
+static size_t FunctionBar_displayWidth(const char *s) {
+   size_t slen = strlen(s);
+   size_t width = 0, i = 0;
+
+   mbtowc(NULL, 0, 0);
+
+   while (i < slen) {
+      wchar_t wch;
+      int mblen = mbtowc(&wch, s + i, slen - i);
+      if (mblen <= 0) {
+         return slen;
+      }
+      i += mblen;
+      int w = wcwidth(wch);
+      if (w > 0) {
+         width += w;
```

Comment:
Don't leave the case of `wcwidth(wch) < 0` out of this.

`wcwidth(wch) < 0` can happen when `!iswprint(wch)` and that means `wch` is a C0 or C1 control character, U+2028 (line separator) or U+2029 (paragraph separator). This is according to [ISO/IEC 30112](https://www.open-std.org/jtc1/sc35/wg5/docs/30112d12.pdf) (look for characters in the `cntrl` category).